### DISCUSSION TITLE
dp.login error: no attribute 'login'

### DISCUSSION THREAD

**Discussion by owolabi35** [datapane/datapane#65] [Help] [answered]
I'm having this error message keeps displaying;
AttributeError: module 'datapane' has no attribute 'login'

What is the way out?
Thank You.

**Comment by mands** [comment]
Can you provide some more information about your configuration as per the other comment - thanks! 

Additionally can you provide some example code that leads up to the error? 

The following should work,

```python
import datapane as dp

dp.login(token=YOUR_TOKEN)
```

**Reply by Davidyang1989** [reply]
I used the template code from Welcome to Datapane Public page and had the same error msg.
Plus does this mean only my persional token works, then how to get one? 
thx

"""
import altair as alt
import pandas as pd
import datapane as dp
from datetime import date

# Sign-in with your unique token
dp.login(token="2a6a....")
"""

**Reply by Davidyang1989** [reply]
I found my token in the setting ,while still the problem

AttributeError: module 'datapane' has no attribute 'login'

**Comment by ymonfet** [comment]
I have the same problem: module 'datapane' has no attribute 'login'
Did you solve this problem?

**Reply by mands** [reply]
Can you provide some more information about your configuration as per the other comment - thanks!

**Comment by mands** [comment]
We're having some difficulties reproducing this error - can anyone who encounters the issue please provide some information about your environment that will help us dig into the issue,

OS:
Python Version:
Python Environment: pip/conda
Using Jupyter: true/false
Datapane Version:

Thanks!

**Comment by ymonfet** [comment]
Hi,

OS: WIndows
Using Jupyter: false
Python Version: 3.8.3
Datapane Version:  0.7.2   print(dp.__version__)



I have installed Datapane with this command:


conda install -c conda-forge datapane


...and the login attribute doesn't exist:

import altair as alt
import pandas as pd
import datapane as dp
from datetime import date

# Sign-in with your unique token
dp.login(token="c1a8...")

AttributeError: module 'datapane' has no attribute 'login

Thanks a lot
Yvan


________________________________
De : Mandeep Gill <notifications@github.com>
Envoyé : 15 février 2021 10:19
À : datapane/datapane <datapane@noreply.github.com>
Cc : ymonfet <ymonfette@hotmail.com>; Comment <comment@noreply.github.com>
Objet : Re: [datapane/datapane] Datapane login (#65)


Can you please provide some information about your environment that will help us dig into the issue,

OS:
Using Jupyter: true/false
Python Version:
Datapane Version:

Thanks!

**Reply by mands** [reply]
Yes, it appears conda has installed an older version of `datapane` - the current version is 0.9.2 . Can you try to update the version of datapane that conda installed (`conda update datapane`), and if any issues occur please post them here. Thanks!

**Comment by mands** [comment, marked as answer]
This issue appears to be related to conda sometimes defaulting to installing an older version of datapane - if you encounter this issue, please try each of the following,

1. `conda update datapane` or `conda update --all`
2. remove datapane, and reinstall the newer version, e.g. `conda install -c conda-forge "datapane>=0.10.0"`
3. creating a new conda env and installing datapane in there

**Reply by Davidyang1989** [reply]
thx for your help 
but 0.9.0 works no more for me as the msg prompts
I updated to 0.10.1 then worked

while I had a new problem - how to share the report to others since there is no link or html report created
but a feed back : ConnectionResetError: [WinError 10054]
is it because I lived in China?

**Reply by Davidyang1989** [reply]
however connect to the website is no a worry  : Connected successfully to https://datapane.com as xxx

**Reply by Davidyang1989** [reply]
problem sovled, I think the internet is the cause, is there any thing can be done to make the connection to the server more smoothly as the user in China？

**Reply by mands** [reply]
Glad to hear it worked for you - yes we're on version 0.10.0 now. Sorry for the internet issues you are facing - our servers are in Europe, but are hosted by Google Cloud and you shouldn't generally have any issues connecting - please keep trying and let us know if you have any more issues.

**Comment by ymonfet** [comment]
Hi,
it worked with conda install -c conda-forge "datapane>=0.9.0"
thanks a lot
Yvan


________________________________
De : Mandeep Gill <notifications@github.com>
Envoyé : 15 février 2021 17:12
À : datapane/datapane <datapane@noreply.github.com>
Cc : ymonfet <ymonfette@hotmail.com>; Comment <comment@noreply.github.com>
Objet : Re: [datapane/datapane] dp.login error: no attribute 'login' (#65)


This issue appears to be related to conda sometimes defaulting to installing an older version of datapane - if you encounter this issue, please try each of the following,

  1.  conda update --all
  2.  remove datapane, and reinstall using the new version, e.g. conda install -c conda-forge "datapane>=0.9.0"
  3.  creating a new conda env and installing datapane in there

**Comment by Davidyang1989** [comment]
Is that possible to partially use servers located in China like Aliyun hosted by Alibaba ?
I assume there would be enormous demand with stabler connection.
Long for it

发送自 Windows 10 版邮件<https://go.microsoft.com/fwlink/?LinkId=550986>应用

发件人: Mandeep Gill<mailto:notifications@github.com>
发送时间: 2021年3月5日 3:20
收件人: datapane/datapane<mailto:datapane@noreply.github.com>
抄送: Davidyang1989<mailto:davidyang1989@hotmail.com>; Comment<mailto:comment@noreply.github.com>
主题: Re: [datapane/datapane] dp.login error: no attribute 'login' (#65)


Glad to hear it worked for you - yes we're on version 0.10.0 now. Sorry for the internet issues you are facing - our servers are in Europe, but are hosted by Google Cloud and you shouldn't generally have any issues connecting - please keep trying and let us know if you have any more issues.

―

**Reply by mands** [reply]
We'll investigate what we can do with Google Cloud to increase our availability across regions including those closer to China to help with the connection stability - hope that helps! Thanks!
